# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webidl/DOMLocalization.webidl
# Commit: 626b40e3aad1
# Full Hash: 626b40e3aad147d59a439b0bd1d5dbd328ad6b1a
# Author: Zibi Braniecki <zbraniecki@mozilla.com>
# Date: 2021-08-02 21:43:56
# Regressor Bug: 1613705
# File Overlap Count: 1
# Description:
#   Bug 1613705 - [localization] part9: Update DOMLocalization to match Localization constructor. r=nika,emilio
#   
#   Depends on D113691
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D113692
# ==============================================================================

diff -r 1cbab987bb7d -r 626b40e3aad1 dom/webidl/DOMLocalization.webidl
--- a/dom/webidl/DOMLocalization.webidl	Mon Aug 02 09:54:10 2021 +0000
+++ b/dom/webidl/DOMLocalization.webidl	Mon Aug 02 09:54:10 2021 +0000
@@ -36,14 +36,14 @@
    *                           This enables a number of synchronous methods on the
    *                           Localization API and uses it for `TranslateElements`
    *                           making the method return a synchronusly resolved promise.
-   *    - aBundleGenerator   - an object with two methods - `generateBundles` and
-   *                           `generateBundlesSync` allowing consumers to overload the
-   *                           default generators provided by Gecko.
+   *    - aRegistry            - optional custom L10nRegistry to be used by this Localization instance.
+   *    - aLocales             - custom set of locales to be used for this Localization.
    */
   [Throws]
-  constructor(sequence<DOMString> aResourceIds,
+  constructor(sequence<UTF8String> aResourceIds,
               optional boolean aSync = false,
-              optional BundleGenerator aBundleGenerator = {});
+              optional L10nRegistry aRegistry,
+              optional sequence<UTF8String> aLocales);
 
   /**
    * Adds a node to nodes observed for localization